Fix build errors in app crate (27 errors across 5 files)

## Summary

The `app` crate (src-tauri) has 27 build errors preventing compilation. These errors span 5 files and fall into several categories that need systematic fixing.

## Error Categories

### 1. Syntax Errors in `optimization.rs` (6 errors)

**Location:** `src-tauri/src/commands/optimization.rs:217-224`

```
error: expected one of `)`, `,`, `.`, `?`, or an operator, found `", "function`
error: expected one of `)`, `,`, `.`, `?`, or an operator, found `", "file`
error: expected one of `)`, `,`, `.`, `?`, or an operator, found `", "error`
error: expected one of `)`, `,`, `.`, `?`, or an operator, found `"]}"`
```

**Likely cause:** Malformed string or macro invocation around line 217.

### 2. Lifetime/Borrowing Errors in `telemetry.rs` (10 errors)

**Location:** `src-tauri/src/commands/telemetry.rs` (lines 194, 209, 224, 238, 380, 393, 468, 483, 498, 512)

```
error[E0597]: `stmt` does not live long enough
```

**Likely cause:** `Statement` objects being used after their borrowed `Connection` goes out of scope. The pattern of binding query results to explicit variables (as done in PR #1102) may need to be applied here as well.

### 3. Lifetime/Borrowing Errors in `ab_testing.rs` (4 errors)

**Location:** `src-tauri/src/commands/ab_testing.rs` (lines 1246, 1253, 1314, 1324)

```
error[E0597]: `conn` does not live long enough
error[E0597]: `stmt` does not live long enough
```

**Likely cause:** Same pattern as telemetry.rs - statements outliving their connections.

### 4. Type Mismatch Errors (4 errors)

**Locations:**
- `ab_testing.rs:699` - `mismatched types`
- `ab_testing.rs:1139` - `if` and `else` have incompatible types
- `telemetry.rs:310` - `if` and `else` have incompatible types
- `telemetry.rs:576` - `if` and `else` have incompatible types

**Likely cause:** Return types not matching between branches or expected vs actual types.

### 5. Missing Chrono Methods in `prediction.rs` (2 errors)

**Location:** `src-tauri/src/commands/prediction.rs` (lines 269, 273)

```
error[E0599]: no method named `hour` found for struct `chrono::DateTime<Tz>`
error[E0599]: no method named `weekday` found for struct `chrono::DateTime<Tz>`
```

**Likely cause:** Missing `use chrono::Timelike;` and `use chrono::Datelike;` imports.

### 6. Ambiguous Numeric Type in `optimization.rs` (1 error)

**Location:** `src-tauri/src/commands/optimization.rs:551`

```
error[E0689]: can't call method `clamp` on ambiguous numeric type `{float}`
```

**Likely cause:** Need explicit type annotation like `(value as f64).clamp(...)`.

## Files Affected

| File | Error Count | Primary Issue |
|------|-------------|---------------|
| `telemetry.rs` | 12 | Lifetime/borrowing + type mismatches |
| `ab_testing.rs` | 6 | Lifetime/borrowing + type mismatches |
| `optimization.rs` | 7 | Syntax errors + ambiguous type |
| `prediction.rs` | 2 | Missing trait imports |

## Acceptance Criteria

- [ ] `cargo build -p app` compiles without errors
- [ ] `cargo test -p app` passes (if tests exist)
- [ ] No new warnings introduced

## Suggested Approach

1. Start with syntax errors in `optimization.rs:217` - these may be blocking other error detection
2. Add missing chrono trait imports to `prediction.rs`
3. Fix type annotations in `optimization.rs:551`
4. Apply explicit result binding pattern (from PR #1102) to fix lifetime errors
5. Fix type mismatches in if/else branches

## Related

- PR #1102 partially addressed similar lifetime issues in `telemetry.rs` but changes were incomplete

--- a/src-tauri/src/commands/ab_testing.rs
+++ b/src-tauri/src/commands/ab_testing.rs
@@ -121,7 +121,7 @@ pub struct Experiment {
 }
 
 /// Variant within an experiment
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Variant {
     pub id: Option<i64>,
     pub experiment_id: i64,
@@ -1119,43 +1119,51 @@ pub fn get_experiments(
         .prepare(query)
         .map_err(|e| format!("Failed to prepare query: {e}"))?;
 
-    let rows = if let Some(ref s) = status {
-        stmt.query_map([s], |row| {
-            Ok(Experiment {
-                id: row.get(0)?,
-                name: row.get(1)?,
-                description: row.get(2)?,
-                hypothesis: row.get(3)?,
-                status: row.get(4)?,
-                created_at: row.get(5)?,
-                started_at: row.get(6)?,
-                concluded_at: row.get(7)?,
-                min_sample_size: row.get(8)?,
-                target_metric: row.get(9)?,
-                target_direction: row.get(10)?,
+    let experiments: Vec<Experiment> = if let Some(ref s) = status {
+        let results: Vec<Experiment> = stmt
+            .query_map([s], |row| {
+                Ok(Experiment {
+                    id: row.get(0)?,
+                    name: row.get(1)?,
+                    description: row.get(2)?,
+                    hypothesis: row.get(3)?,
+                    status: row.get(4)?,
+                    created_at: row.get(5)?,
+                    started_at: row.get(6)?,
+                    concluded_at: row.get(7)?,
+                    min_sample_size: row.get(8)?,
+                    target_metric: row.get(9)?,
+                    target_direction: row.get(10)?,
+                })
             })
-        })
+            .map_err(|e| format!("Failed to query experiments: {e}"))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| format!("Failed to collect experiments: {e}"))?;
+        results
     } else {
-        stmt.query_map([], |row| {
-            Ok(Experiment {
-                id: row.get(0)?,
-                name: row.get(1)?,
-                description: row.get(2)?,
-                hypothesis: row.get(3)?,
-                status: row.get(4)?,
-                created_at: row.get(5)?,
-                started_at: row.get(6)?,
-                concluded_at: row.get(7)?,
-                min_sample_size: row.get(8)?,
-                target_metric: row.get(9)?,
-                target_direction: row.get(10)?,
+        let results: Vec<Experiment> = stmt
+            .query_map([], |row| {
+                Ok(Experiment {
+                    id: row.get(0)?,
+                    name: row.get(1)?,
+                    description: row.get(2)?,
+                    hypothesis: row.get(3)?,
+                    status: row.get(4)?,
+                    created_at: row.get(5)?,
+                    started_at: row.get(6)?,
+                    concluded_at: row.get(7)?,
+                    min_sample_size: row.get(8)?,
+                    target_metric: row.get(9)?,
+                    target_direction: row.get(10)?,
+                })
             })
-        })
+            .map_err(|e| format!("Failed to query experiments: {e}"))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| format!("Failed to collect experiments: {e}"))?;
+        results
     };
 
-    rows.map_err(|e| format!("Failed to query experiments: {e}"))?
-        .collect::<Result<Vec<_>, _>>()
-        .map_err(|e| format!("Failed to collect experiments: {e}"))
+    Ok(experiments)
 }
 
 /// Get a single experiment by ID
@@ -1250,19 +1258,22 @@ pub fn get_experiment_variants(
         )
         .map_err(|e| format!("Failed to prepare query: {e}"))?;
 
-    stmt.query_map([experiment_id], |row| {
-        Ok(Variant {
-            id: row.get(0)?,
-            experiment_id: row.get(1)?,
-            name: row.get(2)?,
-            description: row.get(3)?,
-            config_json: row.get(4)?,
-            weight: row.get(5)?,
+    let results: Vec<Variant> = stmt
+        .query_map([experiment_id], |row| {
+            Ok(Variant {
+                id: row.get(0)?,
+                experiment_id: row.get(1)?,
+                name: row.get(2)?,
+                description: row.get(3)?,
+                config_json: row.get(4)?,
+                weight: row.get(5)?,
+            })
         })
-    })
-    .map_err(|e| format!("Failed to query variants: {e}"))?
-    .collect::<Result<Vec<_>, _>>()
-    .map_err(|e| format!("Failed to collect variants: {e}"))
+        .map_err(|e| format!("Failed to query variants: {e}"))?
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(|e| format!("Failed to collect variants: {e}"))?;
+
+    Ok(results)
 }
 
 /// Get summary of all experiments
@@ -1321,17 +1332,20 @@ pub fn get_experiment_results(
         )
         .map_err(|e| format!("Failed to prepare query: {e}"))?;
 
-    stmt.query_map([experiment_id], |row| {
-        Ok(ExperimentResult {
-            id: row.get(0)?,
-            assignment_id: row.get(1)?,
-            success_factor_id: row.get(2)?,
-            outcome: row.get(3)?,
-            metric_value: row.get(4)?,
-            recorded_at: row.get(5)?,
+    let results: Vec<ExperimentResult> = stmt
+        .query_map([experiment_id], |row| {
+            Ok(ExperimentResult {
+                id: row.get(0)?,
+                assignment_id: row.get(1)?,
+                success_factor_id: row.get(2)?,
+                outcome: row.get(3)?,
+                metric_value: row.get(4)?,
+                recorded_at: row.get(5)?,
+            })
         })
-    })
-    .map_err(|e| format!("Failed to query results: {e}"))?
-    .collect::<Result<Vec<_>, _>>()
-    .map_err(|e| format!("Failed to collect results: {e}"))
+        .map_err(|e| format!("Failed to query results: {e}"))?
+        .collect::<Result<Vec<_>, _>>()
+        .map_err(|e| format!("Failed to collect results: {e}"))?;
+
+    Ok(results)
 }

--- a/src-tauri/src/commands/optimization.rs
+++ b/src-tauri/src/commands/optimization.rs
@@ -214,14 +214,14 @@ fn initialize_default_rules(conn: &Connection) -> SqliteResult<()> {
         (
             "vague_prompt",
             "specificity",
-            r#"{"keywords": ["fix", "update", "change", "modify"], "lacks": ["#", "function", "file", "error"]}"#,
+            r##"{"keywords": ["fix", "update", "change", "modify"], "lacks": ["#", "function", "file", "error"]}"##,
             "Be more specific: Instead of '{{original}}', try: '{{original}} in {{suggested_scope}}'",
             0.20,
         ),
         (
             "missing_issue_ref",
             "structure",
-            r#"{"lacks_pattern": "#[0-9]+"}"#,
+            r##"{"lacks_pattern": "#[0-9]+"}"##,
             "Reference the issue: {{original}} (see issue #N for details)",
             0.12,
         ),
@@ -497,7 +497,7 @@ fn calculate_specificity_score(prompt: &str) -> f64 {
 
 /// Calculate structural quality of a prompt (0.0 to 1.0)
 fn calculate_structure_score(prompt: &str) -> f64 {
-    let mut score = 0.5;
+    let mut score: f64 = 0.5;
 
     let word_count = prompt.split_whitespace().count();
 

--- a/src-tauri/src/commands/prediction.rs
+++ b/src-tauri/src/commands/prediction.rs
@@ -6,6 +6,7 @@
 //!
 //! Part of Phase 4 (Advanced Analytics) - Builds on Phase 3 correlation analysis.
 
+use chrono::{Datelike, Timelike};
 use rusqlite::{params, Connection, Result as SqliteResult};
 use serde::{Deserialize, Serialize};
 use std::path::Path;

--- a/src-tauri/src/commands/telemetry.rs
+++ b/src-tauri/src/commands/telemetry.rs
@@ -180,7 +180,7 @@ pub fn get_performance_metrics(
     let limit = limit.unwrap_or(100);
 
     // Build query and parameters based on which filters are provided
-    let metrics = match (&category, &since) {
+    let metrics: Vec<PerformanceMetric> = match (&category, &since) {
         (Some(cat), Some(since_time)) => {
             let mut stmt = conn
                 .prepare(
@@ -191,10 +191,12 @@ pub fn get_performance_metrics(
                      LIMIT ?3",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![cat, since_time, limit], map_performance_row)
+            let results: Vec<PerformanceMetric> = stmt
+                .query_map(params![cat, since_time, limit], map_performance_row)
                 .map_err(|e| format!("Failed to query metrics: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect metrics: {e}"))?
+                .map_err(|e| format!("Failed to collect metrics: {e}"))?;
+            results
         }
         (Some(cat), None) => {
             let mut stmt = conn
@@ -206,10 +208,12 @@ pub fn get_performance_metrics(
                      LIMIT ?2",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![cat, limit], map_performance_row)
+            let results: Vec<PerformanceMetric> = stmt
+                .query_map(params![cat, limit], map_performance_row)
                 .map_err(|e| format!("Failed to query metrics: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect metrics: {e}"))?
+                .map_err(|e| format!("Failed to collect metrics: {e}"))?;
+            results
         }
         (None, Some(since_time)) => {
             let mut stmt = conn
@@ -221,10 +225,12 @@ pub fn get_performance_metrics(
                      LIMIT ?2",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![since_time, limit], map_performance_row)
+            let results: Vec<PerformanceMetric> = stmt
+                .query_map(params![since_time, limit], map_performance_row)
                 .map_err(|e| format!("Failed to query metrics: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect metrics: {e}"))?
+                .map_err(|e| format!("Failed to collect metrics: {e}"))?;
+            results
         }
         (None, None) => {
             let mut stmt = conn
@@ -235,10 +241,12 @@ pub fn get_performance_metrics(
                      LIMIT ?1",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![limit], map_performance_row)
+            let results: Vec<PerformanceMetric> = stmt
+                .query_map(params![limit], map_performance_row)
                 .map_err(|e| format!("Failed to query metrics: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect metrics: {e}"))?
+                .map_err(|e| format!("Failed to collect metrics: {e}"))?;
+            results
         }
     };
 
@@ -294,33 +302,39 @@ pub fn get_performance_stats(since: Option<String>) -> Result<Vec<PerformanceSta
         .prepare(query)
         .map_err(|e| format!("Failed to prepare query: {e}"))?;
 
-    let stats = if let Some(since_time) = since {
-        stmt.query_map([since_time], |row| {
-            Ok(PerformanceStats {
-                category: row.get(0)?,
-                count: row.get(1)?,
-                avg_duration_ms: row.get(2)?,
-                max_duration_ms: row.get(3)?,
-                min_duration_ms: row.get(4)?,
-                success_rate: row.get(5)?,
+    let stats: Vec<PerformanceStats> = if let Some(since_time) = since {
+        let results: Vec<PerformanceStats> = stmt
+            .query_map([since_time], |row| {
+                Ok(PerformanceStats {
+                    category: row.get(0)?,
+                    count: row.get(1)?,
+                    avg_duration_ms: row.get(2)?,
+                    max_duration_ms: row.get(3)?,
+                    min_duration_ms: row.get(4)?,
+                    success_rate: row.get(5)?,
+                })
             })
-        })
-        .map_err(|e| format!("Failed to query stats: {e}"))?
+            .map_err(|e| format!("Failed to query stats: {e}"))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| format!("Failed to collect stats: {e}"))?;
+        results
     } else {
-        stmt.query_map([], |row| {
-            Ok(PerformanceStats {
-                category: row.get(0)?,
-                count: row.get(1)?,
-                avg_duration_ms: row.get(2)?,
-                max_duration_ms: row.get(3)?,
-                min_duration_ms: row.get(4)?,
-                success_rate: row.get(5)?,
+        let results: Vec<PerformanceStats> = stmt
+            .query_map([], |row| {
+                Ok(PerformanceStats {
+                    category: row.get(0)?,
+                    count: row.get(1)?,
+                    avg_duration_ms: row.get(2)?,
+                    max_duration_ms: row.get(3)?,
+                    min_duration_ms: row.get(4)?,
+                    success_rate: row.get(5)?,
+                })
             })
-        })
-        .map_err(|e| format!("Failed to query stats: {e}"))?
-    }
-    .collect::<Result<Vec<_>, _>>()
-    .map_err(|e| format!("Failed to collect stats: {e}"))?;
+            .map_err(|e| format!("Failed to query stats: {e}"))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| format!("Failed to collect stats: {e}"))?;
+        results
+    };
 
     Ok(stats)
 }
@@ -367,7 +381,7 @@ pub fn get_error_reports(
     let limit = limit.unwrap_or(50);
 
     // Build query and parameters based on whether since filter is provided
-    let errors = if let Some(since_time) = since {
+    let errors: Vec<ErrorReport> = if let Some(since_time) = since {
         let mut stmt = conn
             .prepare(
                 "SELECT message, stack, timestamp, component, context
@@ -377,10 +391,12 @@ pub fn get_error_reports(
                  LIMIT ?2",
             )
             .map_err(|e| format!("Failed to prepare query: {e}"))?;
-        stmt.query_map(params![since_time, limit], map_error_row)
+        let results: Vec<ErrorReport> = stmt
+            .query_map(params![since_time, limit], map_error_row)
             .map_err(|e| format!("Failed to query errors: {e}"))?
             .collect::<Result<Vec<_>, _>>()
-            .map_err(|e| format!("Failed to collect errors: {e}"))?
+            .map_err(|e| format!("Failed to collect errors: {e}"))?;
+        results
     } else {
         let mut stmt = conn
             .prepare(
@@ -390,10 +406,12 @@ pub fn get_error_reports(
                  LIMIT ?1",
             )
             .map_err(|e| format!("Failed to prepare query: {e}"))?;
-        stmt.query_map(params![limit], map_error_row)
+        let results: Vec<ErrorReport> = stmt
+            .query_map(params![limit], map_error_row)
             .map_err(|e| format!("Failed to query errors: {e}"))?
             .collect::<Result<Vec<_>, _>>()
-            .map_err(|e| format!("Failed to collect errors: {e}"))?
+            .map_err(|e| format!("Failed to collect errors: {e}"))?;
+        results
     };
 
     Ok(errors)
@@ -454,7 +472,7 @@ pub fn get_usage_events(
     let limit = limit.unwrap_or(100);
 
     // Build query and parameters based on which filters are provided
-    let events = match (&category, &since) {
+    let events: Vec<UsageEvent> = match (&category, &since) {
         (Some(cat), Some(since_time)) => {
             let mut stmt = conn
                 .prepare(
@@ -465,10 +483,12 @@ pub fn get_usage_events(
                      LIMIT ?3",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![cat, since_time, limit], map_usage_row)
+            let results: Vec<UsageEvent> = stmt
+                .query_map(params![cat, since_time, limit], map_usage_row)
                 .map_err(|e| format!("Failed to query events: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect events: {e}"))?
+                .map_err(|e| format!("Failed to collect events: {e}"))?;
+            results
         }
         (Some(cat), None) => {
             let mut stmt = conn
@@ -480,10 +500,12 @@ pub fn get_usage_events(
                      LIMIT ?2",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![cat, limit], map_usage_row)
+            let results: Vec<UsageEvent> = stmt
+                .query_map(params![cat, limit], map_usage_row)
                 .map_err(|e| format!("Failed to query events: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect events: {e}"))?
+                .map_err(|e| format!("Failed to collect events: {e}"))?;
+            results
         }
         (None, Some(since_time)) => {
             let mut stmt = conn
@@ -495,10 +517,12 @@ pub fn get_usage_events(
                      LIMIT ?2",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![since_time, limit], map_usage_row)
+            let results: Vec<UsageEvent> = stmt
+                .query_map(params![since_time, limit], map_usage_row)
                 .map_err(|e| format!("Failed to query events: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect events: {e}"))?
+                .map_err(|e| format!("Failed to collect events: {e}"))?;
+            results
         }
         (None, None) => {
             let mut stmt = conn
@@ -509,10 +533,12 @@ pub fn get_usage_events(
                      LIMIT ?1",
                 )
                 .map_err(|e| format!("Failed to prepare query: {e}"))?;
-            stmt.query_map(params![limit], map_usage_row)
+            let results: Vec<UsageEvent> = stmt
+                .query_map(params![limit], map_usage_row)
                 .map_err(|e| format!("Failed to query events: {e}"))?
                 .collect::<Result<Vec<_>, _>>()
-                .map_err(|e| format!("Failed to collect events: {e}"))?
+                .map_err(|e| format!("Failed to collect events: {e}"))?;
+            results
         }
     };
 
@@ -562,29 +588,35 @@ pub fn get_usage_stats(since: Option<String>) -> Result<Vec<UsageStats>, String>
         .prepare(query)
         .map_err(|e| format!("Failed to prepare query: {e}"))?;
 
-    let stats = if let Some(since_time) = since {
-        stmt.query_map([since_time], |row| {
-            Ok(UsageStats {
-                event_name: row.get(0)?,
-                category: row.get(1)?,
-                count: row.get(2)?,
-                last_occurrence: row.get(3)?,
+    let stats: Vec<UsageStats> = if let Some(since_time) = since {
+        let results: Vec<UsageStats> = stmt
+            .query_map([since_time], |row| {
+                Ok(UsageStats {
+                    event_name: row.get(0)?,
+                    category: row.get(1)?,
+                    count: row.get(2)?,
+                    last_occurrence: row.get(3)?,
+                })
             })
-        })
-        .map_err(|e| format!("Failed to query stats: {e}"))?
+            .map_err(|e| format!("Failed to query stats: {e}"))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| format!("Failed to collect stats: {e}"))?;
+        results
     } else {
-        stmt.query_map([], |row| {
-            Ok(UsageStats {
-                event_name: row.get(0)?,
-                category: row.get(1)?,
-                count: row.get(2)?,
-                last_occurrence: row.get(3)?,
+        let results: Vec<UsageStats> = stmt
+            .query_map([], |row| {
+                Ok(UsageStats {
+                    event_name: row.get(0)?,
+                    category: row.get(1)?,
+                    count: row.get(2)?,
+                    last_occurrence: row.get(3)?,
+                })
             })
-        })
-        .map_err(|e| format!("Failed to query stats: {e}"))?
-    }
-    .collect::<Result<Vec<_>, _>>()
-    .map_err(|e| format!("Failed to collect stats: {e}"))?;
+            .map_err(|e| format!("Failed to query stats: {e}"))?
+            .collect::<Result<Vec<_>, _>>()
+            .map_err(|e| format!("Failed to collect stats: {e}"))?;
+        results
+    };
 
     Ok(stats)
 }

